jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -1320,17 +1323,19 @@ else if (last && !(totalWritten == 0 && HttpMethod.HEAD.is(_request.getMethod())
                     return;
                 }
 
+                // TODO: check that we don't have both a ByteBuffer and a Content.Source.
```

Comment:
I think you should also check `content==TRANSFER`, because if some sink wrapper inserted different content, then the presence of the source will need to be ignored